### PR TITLE
VPN-3681 - add datetime metric to glean

### PR DESCRIPTION
**Pull request by mcleinman**
## Description

Adding datetime metric to glean. From some local tests, it seems to work.

While most of the `datetime` object is covering in this PR, two pieces were left for the future (if they're ever needed). This work is captured [in a ticket](https://mozilla-hub.atlassian.net/browse/VPN-4173), as well as a comment within `datetime.h`.

## Reference

https://mozilla-hub.atlassian.net/browse/VPN-3681

## Checklist
    
- [X] My code follows the style guidelines for this project
- [X] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [X] I have performed a self review of my own code
- [X] I have commented my code PARTICULARLY in hard to understand areas
- [X] I have added thorough tests where needed

### CODE REVIEW

File: qtglean/include/glean/datetime.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+#ifndef DATETIME_H
+#define DATETIME_H
+#include <QObject>
+#include <QString>
+
+#include "errortype.h"
+
+// NOTE: While most of datetime is implemented, two pieces are not yet:
+// - The ability to set an arbitrary datestamp
+// - testGetValue, which provides a language-specific object
```
[the remark this answers is not in the document]
Comment (mcleinman) [reply qtglean/include/glean/datetime.h]: Good point - I added `testGetValue`.

I have code for the arbitrary datestamp, but we'd need to bring the chrono library into rust to work w/ date stamps. A very quick internet glance seems to imply it was the standard, and possibly went dormant for a bit, and possibly now has a new maintainer - even if half of that is true, is it worth bringing in the library for something we don't currently need?

[the remark this answers is not in the document]
Comment (mcleinman) [reply qtglean/include/glean/datetime.h]: Alternatively could do the [fog approach](https://searchfox.org/mozilla-central/source/toolkit/components/glean/api/src/ffi/datetime.rs) and create a new struct.

[the remark this answers is not in the document]
Comment (brizental) [reply qtglean/include/glean/datetime.h]: Hum, I'm fine delaying the API to set the arbitrary datetime if that is not actually trivial :)

[the remark this answers is not in the document]
Comment (brizental) [reply qtglean/include/glean/datetime.h]: Resolving because I see you have already updated it :100:

File: qtglean/include/glean/datetime.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+#ifndef DATETIME_H
+#define DATETIME_H
+#include <QDateTime>
+#include <QObject>
+#include <QString>
+
+#include "errortype.h"
+
+// NOTE: While most of datetime is implemented, one pieces are not yet:
```
Comment (brizental) [inline qtglean/include/glean/datetime.h:14]: super nit: 
```suggestion
// NOTE: While most of datetime is implemented, one piece is not yet:
```

Comment (mcleinman) [reply qtglean/include/glean/datetime.h]: Good catch, thanks.

**Pull request merged**